icons/icons.json

```diff
@@ -3744,5 +3743,139 @@
 				"scope": "source.zig"
 			}
 		]
+	},
+	"file_type_esbuild": {
+		"aliases": [
+			{
+				"base": "source.js",
+				"extensions": [
+					"esbuild.mjs",
+					"esbuild.cjs",
+					"esbuild.js",
+					"esbuild.config.js",
+					"esbuild.config.cjs",
+					"esbuild.config.mjs",
+					"esbuild.prod.js",
+					"esbuild.prod.mjs",
+					"esbuild.prod.cjs",
+					"esbuild.dev.js",
+					"esbuild.dev.mjs",
+					"esbuild.dev.cjs"
+				],
+				"name": "JavaScript (ESBuild)",
+				"scope": "source.js.esbuild"
+			},
+			{
+				"base": "source.ts",
+				"extensions": [
+					"esbuild.ts",
+					"esbuild.cts",
+					"esbuild.mts",
+					"esbuild.config.ts",
+					"esbuild.config.cts",
+					"esbuild.config.mts",
+					"esbuild.prod.ts",
+					"esbuild.prod.mts",
+					"esbuild.prod.cts",
+					"esbuild.dev.ts",
+					"esbuild.dev.mts",
+					"esbuild.dev.cts"
+				],
+				"name": "TypeScript (ESBuild)",
+				"scope": "source.ts.esbuild"
+			}
+		],
+		"color": "yellow"
+	},
+	"file_type_jsconfig": {
+		"aliases": [
+			{
+				"base": "source.json",
+				"extensions": [
+					"jsconfig.json"
+				],
+				"name": "JSON (JSConfig)",
+				"scope": "source.json.jsconfig"
+			}
+		],
+		"color": "yellow"
+	},
+	"file_type_tsconfig": {
+		"aliases": [
+			{
+				"base": "source.json",
+				"extensions": [
+					"tsconfig.json"
+				],
+				"name": "JSON (TSConfig)",
+				"scope": "source.json.tsconfig"
+			}
+		],
+		"color": "blue"
+	},
+	"file_type_makefile": {
+		"aliases": [
+			{
+				"base": "source.makefile",
+				"extensions": [
+					"makefile"
+				],
+				"name": "Makefile",
+				"scope": "source.makefile"
+			}
+		],
+		"color": "red"
+	},
+	"file_type_readme": {
```

Comment:
A readme somehow follows already existing `file_type_license`. There's however also a `file_type_text` which covers files such as `README`.

Finally readmes can be of any type. Also RST or anything along those lines.

We could try to add dozens of aliases to catch them all, but finally this would augment their type. Not sure if it is worth it.